File: server/php/src/Infrastructure/Mappers/PaymentDetails/PaymentProduct840SpecificOutputMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\PaymentProduct840SpecificOutput as PaymentProduct840SpecificOutputDto;
use OnlinePayments\Sdk\Domain\PaymentProduct840SpecificOutput as PaymentProduct840SpecificOutputSdk;

final class PaymentProduct840SpecificOutputMapper
{
    public static function mapFromSdkResponse(?PaymentProduct840SpecificOutputSdk $response): ?PaymentProduct840SpecificOutputDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new PaymentProduct840SpecificOutputDto();
        $dto->billingAddress = AddressMapper::mapFromSdkResponse($response?->billingAddress ?? null);
        $dto->customerAccount = PaymentProduct840CustomerAccountMapper::mapFromSdkResponse($response?->customerAccount ?? null);
        $dto->customerAddress = AddressMapper::mapFromSdkResponse($response?->customerAddress ?? null);
        $dto->protectionEligibility = ProtectionEligibilityMapper::mapFromSdkResponse($response?->protectionEligibility ?? null);

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/PaymentProduct840SpecificOutputMapper.cs
using PaymentProduct840SpecificOutputSdk = OnlinePayments.Sdk.Domain.PaymentProduct840SpecificOutput;
using PaymentProduct840SpecificOutputDto = Business.Domain.Payments.PaymentDetails.PaymentProduct840SpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class PaymentProduct840SpecificOutputMapper
{
    public static PaymentProduct840SpecificOutputDto Map(PaymentProduct840SpecificOutputSdk? response)
    {
        return new()
        {
            BillingAddress = AddressMapper.Map(response?.BillingAddress),
            CustomerAccount = PaymentProduct840CustomerAccountMapper.Map(response?.CustomerAccount),
            CustomerAddress = AddressMapper.Map(response?.CustomerAddress),
            ProtectionEligibility = ProtectionEligibilityMapper.Map(response?.ProtectionEligibility)
        };
    }
}